edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookService.java

```diff
@@ -683,16 +683,15 @@ public String getAssignmentScoreString(String gradebookUid, Long assignmentId, S
 	
 	/**
 	 * Get student's assignment's score as string.
-	 * This is provided for backward compatibility only.
+	 * This method will attempt to lookup the Long value if provided as a String
+	 * You should use {@link #getAssignmentScoreString(String, Long, String)}
 	 * 
 	 * @param gradebookUid
 	 * @param assignmentName
 	 * @param studentUid
 	 * @return String of score
 	 * 
-	 * @deprecated See {@link #getAssignmentScoreString(String, Long, String)}
 	 */
-	@Deprecated
 	public String getAssignmentScoreString(String gradebookUid, String assignmentName, String studentUid)
```

Comment:
I think we should keep these deprecated. I am actually not sure how safe this conversion is. For the reasons you outline, if someone creates an assignment with 1 as the title, it will get the assignment with ID 1 and potentially the wrong grade. What if the number lookup was a fallback option instead of the primary?